doc/api/permissions.md

```diff
@@ -440,7 +446,152 @@ not adopt the origin of the `blob:` URL.
+will restrict access to all available permissions.
+
+The available permissions are:
+
+* File System - managed through the [`--allow-fs`][] flag
+* Child Process - managed through the [`--allow-child-process`][] flag
+* Worker Threads - managed through the [`--allow-worker`][] flag
+
+When starting Node.js with `--experimental-permission`,
+the ability to access the filesystem, spawn processes, and
+use `node:worker_threads` will be restricted.
+
+```console
+$ node --experimental-permission index.js
+node:internal/modules/cjs/loader:171
+  const result = internalModuleStat(filename);
+                 ^
+
+Error: Access to this API has been restricted
+    at stat (node:internal/modules/cjs/loader:171:18)
+    at Module._findPath (node:internal/modules/cjs/loader:627:16)
+    at resolveMainPath (node:internal/modules/run_main:19:25)
+    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:76:24)
+    at node:internal/main/run_main_module:23:47 {
+  code: 'ERR_ACCESS_DENIED',
+  permission: 'FileSystemRead'
+}
+```
+
+Allowing access to spawning a process and creating worker threads can be done
+using the `--allow-child-process` and `--allow-worker` respectively.
+
+#### Runtime API
+
+When enabling the Permission Model through the [`--experimental-permission`][]
+flag a new property `permission` is added to the `process` object.
+This property contains two functions:
+
+* `permission.deny(scope [,parameters])`
+
+API call to deny permissions at runtime ([`permission.deny()`][])
+
+```js
+process.permission.deny('fs'); // Deny permissions to ALL fs operations
+
+// Deny permissions to ALL FileSystemWrite operations
+process.permission.deny('fs.write');
+// deny FileSystemWrite permissions to the protected-folder
+process.permission.deny('fs.write', ['/home/rafaelgss/protected-folder']);
+// Deny permissions to ALL FileSystemRead operations
+process.permission.deny('fs.read');
+// deny FileSystemRead permissions to the protected-folder
+process.permission.deny('fs.read', ['/home/rafaelgss/protected-folder']);
+```
+
+API call to check permissions at runtime ([`permission.has()`][])
+
+```js
+process.permission.has('fs.write'); // true
+process.permission.has('fs.write', '/home/rafaelgss/protected-folder'); // true
+
+process.permission.deny('fs.write', '/home/rafaelgss/protected-folder');
+
+process.permission.has('fs.write'); // true
+process.permission.has('fs.write', '/home/rafaelgss/protected-folder'); // false
+```
+
+#### File System Permissions
+
+To allow access to the filesystem, use the flag `--allow-fs`:
+
+```console
+$ node --experimental-permission --allow-fs=read index.js
+Hello world!
+(node:19836) ExperimentalWarning: Permission is an experimental feature
+(Use `node --trace-warnings ...` to show where the warning was created)
+```
+
+The valid arguments for the `--allow-fs` flag option are:
+
+* `write` - To manage `FileSystemWrite` (writing) operations.
```

Comment:
It should be documented which operations require write access to which part of the file system. For example, IIRC, on many operating systems, creating a new file in a directory requires write permissions on the directory itself, whereas opening the same file for writing requires write permissions on the file but not on the directory. As far as I can tell, this implementation follows its own set of rules.

Reply:
Sorry, I don't understand. 

> creating a new file in a directory requires write permissions on the directory itself, whereas opening the same file for writing requires write permissions on the file but not on the directory

This is correct. The permission model works this way. Do you mean documenting cases like: if you deny read access to everything you won't be able to require a module?